src/Compile.hs

```diff
@@ -28,85 +28,98 @@ class Compilable a where
+      Solved _ _ (App (Solved _ _ (Var "%")) arg' _) ->
         compile astPath outputPath arg'
           <> " % "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var "==")) arg') ->
-        compile astPath outputPath arg'
-          <> " === "
+      Solved _ _ (App (Solved _ _ (Var "==")) arg' _) ->
+        "__eq("
+          <> compile astPath outputPath arg'
+          <> ", "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var "&&")) arg') ->
+          <> ")"
+      -- Solved _ _ (App (Solved _ _ (Var "==")) arg') ->
+      --   compile astPath outputPath arg'
+      --     <> " === "
+      --     <> compile astPath outputPath arg
+      Solved _ _ (App (Solved _ _ (Var "&&")) arg' _) ->
         compile astPath outputPath arg'
           <> " && "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var "||")) arg') ->
+      Solved _ _ (App (Solved _ _ (Var "||")) arg' _) ->
         compile astPath outputPath arg'
           <> " || "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var ">")) arg') ->
+      Solved _ _ (App (Solved _ _ (Var ">")) arg' _) ->
         compile astPath outputPath arg'
           <> " > "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var "<")) arg') ->
+      Solved _ _ (App (Solved _ _ (Var "<")) arg' _) ->
         compile astPath outputPath arg'
           <> " < "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var ">=")) arg') ->
+      Solved _ _ (App (Solved _ _ (Var ">=")) arg' _) ->
         compile astPath outputPath arg'
           <> " >= "
           <> compile astPath outputPath arg
-      Solved _ _ (App (Solved _ _ (Var "<=")) arg') ->
+      Solved _ _ (App (Solved _ _ (Var "<=")) arg' _) ->
         compile astPath outputPath arg'
           <> " <= "
           <> compile astPath outputPath arg
 
-      Solved _ _ (App (Solved _ _ (Var "|>")) arg') ->
+      Solved _ _ (App (Solved _ _ (Var "|>")) arg' _) ->
         compile astPath outputPath arg
           <> "("
           <> compile astPath outputPath arg'
           <> ")"
 
-      _ -> compileApp [] abs arg
+      _ -> compileApp [] [] abs arg final
      where
-      compileApp :: [String] -> Exp -> Exp -> String
-      compileApp prevArgs abs arg =
-        let
-          args = [compile astPath outputPath arg] <> prevArgs
-          next = case abs of
-            (Solved _ _ (App abs' arg')) -> compileApp args abs' arg'
-            _ ->
-              compile astPath outputPath abs
-                <> "("
-                <> intercalate ", " args
-                <> ")"
-        in
-          next
+      compileApp :: [Bool] -> [String] -> Exp -> Exp -> Bool -> String
+      compileApp prevFinals prevArgs abs arg final =
+        let args   = [compile astPath outputPath arg] <> prevArgs
+            finals = [final] <> prevFinals
```

Comment:
So mostly I build an array of Bool in the recursion that follows the args applied to end up with an array of tuples `(String, Bool)` where the string is the compiled arg ( it could be any exp really ), and the Bool is that final value to know where application stops in the original source code.